manifests/install.pp

```diff
@@ -131,12 +131,13 @@
     }
 
     if $_install_status == 'upgrade' {
-      $_current_version = $::facts['virtualbox_version'].match(/^5[^0-9]{1}[0-9]{1}/)
+      $_current_version = $::facts['virtualbox_version'].match(/^[0-9][^0-9]{1}[0-9]{1}/)
 
       package { "${package_name}-${_current_version[0]}":
         ensure => absent,
-        notify => Package['virtualbox'],
       }
+
+      Package["${package_name}-${_current_version[0]}"] -> Package['virtualbox']
```

Comment:
Why do this over a notify? The ordering metaparameters should be used before arrow syntax.

Reply:
Originally, i used the notify, but some times the order was not good during acceptance runs. With this method i always got the right order.

Reply:
Really?  I have my doubts.  The compiled catalog should be identical with both styles.